src/plugins/data_views/common/fields/field_list.ts

```diff
@@ -68,6 +68,8 @@ export interface IIndexPatternFieldList extends Array<DataViewField> {
    * @return map of field specs by name
    */
   toSpec(options?: ToSpecOptions): DataViewFieldMap;
+
+  hasEcsFields(): boolean;
```

Comment:
These changes should be created on AbstractDataView instead as the field list isn't part of DataViewLazy

Reply:
thx @mattkime need to think about this. the `hasEcsFields` function is tied to the existence of a field list and needs ecs.version to be present. this isn't the case for DataViewLazy, so if we want to use it there is has to work differently

Reply:
I've removed that because apart form the potential `AbstractDataView` case, Lens also works differently ... I've tried out an alternative approach.. follow up in the new code